docs/DebuggingTheCompiler.rst

```diff
@@ -299,3 +299,75 @@ function in the current frame::
     Summary: CollectionType3`ext.CollectionType3.CollectionType3.MutableCollectionType2<A where A: CollectionType3.MutableCollectionType2>.(subscript.materializeForSet : (Swift.Range<A.Index>) -> Swift.MutableSlice<A>).(closure #1)
     Module: file = "/Volumes/Files/work/solon/build/build-swift/validation-test-macosx-x86_64/stdlib/Output/CollectionType.swift.gyb.tmp/CollectionType3", arch = "x86_64"
     Symbol: id = {0x0000008c}, range = [0x0000000100004db0-0x00000001000056f0), name="ext.CollectionType3.CollectionType3.MutableCollectionType2<A where A: CollectionType3.MutableCollectionType2>.(subscript.materializeForSet : (Swift.Range<A.Index>) -> Swift.MutableSlice<A>).(closure #1)", mangled="_TFFeRq_15CollectionType322MutableCollectionType2_S_S0_m9subscriptFGVs5Rangeqq_s16MutableIndexable5Index_GVs12MutableSliceq__U_FTBpRBBRQPS0_MS4__T_"
+
+Debugging failures in LLDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes one needs to be able to while debugging actually debug LLDB and its
+interaction with Swift itself. Some examples of problems where this can come up
+are:
+
+1. Compiler bugs when LLDB attempts to evaluate an expression. (expression
+   debugging)
+2. Swift variables being shown with no types. (type debugging)
+
+To gain further insight into these sorts of failures, we use LLDB log
+channels. LLDB log channels provide introspection by causing LLDB to dump
+verbose information into the log channels as it works. The two log channels that
+are useful for debugging Swift issues are the "types" and "expression" log
+channels.
+
+For more details about any of the information below, please run::
+
+    (lldb) help log enable
+
+"Types" Log
+```````````
+
+The "types" log reports on LLDB's process of:
+
+1. Constructing a SwiftASTContext.
+2. Importing Swift modules from your program into the SwiftASTContext.
+   importing the Swift modules in the program into it.
+
+A "types" error occurs when LLDB fails to import said Swift modules into the
+SwiftASTContext resulting in errors such as swift variables not having
+types. This error can be debugged by turning on the types log::
+
+    (lldb) log enable -f /tmp/lldb-types-log.txt lldb types
+
+That will write the types log to the file passed to the -f option.
+
+**NOTE** It is often times not clear at what point modules are actually loaded by LLDB. To work around such issues, place this command into either::
```

Comment:
Saying "it is often times not clear" makes this sound indeterminate.  I'd say:

module loading can happen as a side-effect of other operations in lldb (e.g. the "file" command) so to be sure you capture all the relevant logs, place this command into either: